core/services/llo/keyring.go

```diff
@@ -0,0 +1,78 @@
+package llo
+
+import (
+	"fmt"
+
+	"github.com/smartcontractkit/libocr/offchainreporting2/types"
+	"github.com/smartcontractkit/libocr/offchainreporting2plus/ocr3types"
+	ocrtypes "github.com/smartcontractkit/libocr/offchainreporting2plus/types"
+
+	commontypes "github.com/smartcontractkit/chainlink-common/pkg/types"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+)
+
+type LLOOnchainKeyring ocr3types.OnchainKeyring[commontypes.LLOReportInfo]
+
+var _ LLOOnchainKeyring = &onchainKeyring{}
+
+type Key interface {
+	Sign3(digest ocrtypes.ConfigDigest, seqNr uint64, r ocrtypes.Report) (signature []byte, err error)
```

Comment:
what is the significance of the trailing `3`? why can't it just be `Sign`?

Reply:
Because `Sign` already refers to the OCR2 signing. `Sign3` is for OCR3. See also: https://github.com/smartcontractkit/chainlink/pull/11956

This could use some followup work to clean it up, that is scheduled here: https://smartcontract-it.atlassian.net/browse/MERC-3594